### ISSUE TITLE
bug(auth): LOGIN action not dispatched after currentUser.linkWithCredential() 

### ISSUE THREAD

**Issue by adrienjt** [prescottprue/react-redux-firebase#268]
Hi,

I use `react-redux-firebase@2.0.0-beta.8` with `react-native-firebase@2.1.4`. Users can start using the application anonymously and later log in with Facebook (keeping the anonymous `uid` if the linked Facebook account isn't already a user).

Here are snippets from thunks:

1. Login anonymously (from unauth state):
    ```javascript
    await firebase.auth().signInAnonymously();
    ```
1. Login with Facebook (from unauth state):
    ```javascript
    await firebase.auth().signInWithCredential(credential);
    ```
1. Logout
    ```javascript
    await firebase.auth().signOut();
    ```
1. Continue with Facebook after logging in anonymously:
    ```javascript
    const {currentUser} = firebase.auth();
    await currentUser.linkWithCredential(credential);
    ```

1, 2, and 3 work as expected, but 4 doesn't dispatch `@@reactReduxFirebase/LOGIN`, therefore `firebase.auth` isn't updated, even though the operation was successful. (If I reload the app, an initial `@@reactReduxFirebase/LOGIN` is dispatched and `firebase.auth` isn't anonymous anymore.)

As a workaround, I manually dispatch the action:
```javascript
const {currentUser} = firebase.auth();
const user = await currentUser.linkWithCredential(credential);
dispatch({type: constants.actionTypes.LOGIN, auth: user});
```

Here is the detail of the relevant dependencies and versions:
```javascript
{
    "react": "15.4.2",
    "react-native": "0.41.2",
    "react-native-fbsdk": "0.5.0",
    "react-native-firebase": "2.1.4",
    "react-redux": "4.4.5",
    "react-redux-firebase": "2.0.0-beta.8",
    "redux": "3.6.0",
    "redux-devtools-extension": "2.13.2",
    "redux-persist": "4.0.0-beta1",
    "redux-thunk": "2.1.0"
}
```

Great library by the way! I'm so glad I don't have to maintain login state myself.

**Comment by prescottprue** [comment]
Will look into reproducing this later today, thanks for reporting!

**Comment by prescottprue** [comment]
@adrienjt I was able to reproduce, so a fix for this is now in the works (basically what you did with dispatching the action). Should come to v2 soon, might take a little longer to make it back to the `1.*.*` versions.

**Comment by prescottprue** [comment]
Planning on adding a `linkWithCredential` method to the firebaseInstance that should handle this. Should hopefully be part of [`v2.0.0-beta.9`](https://github.com/prescottprue/react-redux-firebase/pull/281).

**Comment by prescottprue** [comment]
[`linkWithCredential` is included in `v2.0.0-beta.9` release](https://github.com/prescottprue/react-redux-firebase/releases/tag/v2.0.0-beta.9)
